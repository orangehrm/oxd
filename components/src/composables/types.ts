import {InjectionKey, Ref} from 'vue';

export type ErrorField = {
  cid: string;
  errors: string[];
};

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export type Rule = (value?: any) => string | boolean;
// eslint-disable-next-line @typescript-eslint/no-explicit-any
export type ModelValue = Ref<any>;

export type ErrorBag = Array<ErrorField>;
export type Fieldset = Array<FormField>;

export interface FormField {
  cid: Ref<string>;
  label: Ref<string>;
  dirty: Ref<boolean>;
  touched: Ref<boolean>;
  processing: Ref<boolean>;
  validate: () => Promise<ErrorField>;
  reset: () => void;
}

export interface FormAPI {
  searchErrors: (id: string) => ErrorField[];
  purgeErrors: (id: string) => void;
  addError: (errorField: ErrorField) => void;
  registerField: (field: FormField) => void;
  unregisterField: (field: FormField) => void;
}

export const formKey: InjectionKey<FormAPI> = Symbol('form');
