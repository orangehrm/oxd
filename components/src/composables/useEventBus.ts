interface EventQueue {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  [key: string]: Array<(data: any) => void>;
}

class EventBus {
  private queue: EventQueue;
  constructor() {
    this.queue = {};
  }

  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  $on(name: string, callback: (data: any) => void) {
    this.queue[name] = this.queue[name] || [];
    this.queue[name].push(callback);
  }

  $off(name: string, callback: () => void) {
    if (this.queue[name]) {
      for (let i = 0; i < this.queue[name].length; i++) {
        if (this.queue[name][i] === callback) {
          this.queue[name].splice(i, 1);
          break;
        }
      }
    }
  }

  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  $emit(name: string, data: any) {
    if (this.queue[name]) {
      this.queue[name].forEach(function(callback) {
        callback(data);
      });
    }
  }
}

export default function useEventBus() {
  return new EventBus();
}
