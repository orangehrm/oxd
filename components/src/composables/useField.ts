/*
 * This file is part of OrangeHRM Inc
 *
 * Copyright (C) 2020 onwards OrangeHRM Inc
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see  http://www.gnu.org/licenses
 */

import {ref, Ref, watch, computed, onBeforeUnmount, WatchStopHandle} from 'vue';
import {nanoid} from 'nanoid';
import {injectStrict} from '../utils/injectable';
import {ErrorField, FormAPI, formKey, rule} from './types';

export default function useField(fieldContext: {
  fieldLabel: string;
  rules: rule[];
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  modelValue: Ref<any>;
  onReset: () => Promise<void>;
  disabled: Ref<boolean>;
}) {
  const form = injectStrict<FormAPI>(formKey);
  const cid = ref<string>(nanoid());
  const label = ref<string>(fieldContext.fieldLabel);
  const dirty = ref<boolean>(false);
  const touched = ref<boolean>(false);
  const processing = ref<boolean>(false);
  let watchHandler: WatchStopHandle;

  const validate = () => {
    if (fieldContext.disabled.value) {
      return Promise.resolve({cid: cid.value, errors: []});
    }

    processing.value = true;
    const allValidations = Promise.all(
      fieldContext.rules.map(func => {
        return new Promise<boolean>((resolve, reject) => {
          Promise.resolve(func(fieldContext.modelValue.value)).then(valid => {
            if (valid === true) {
              resolve(valid);
            } else if (typeof valid === 'string') {
              reject(valid);
            } else {
              reject(
                new Error(
                  `Rules should return a string or true, received '${typeof valid}'`,
                ),
              );
            }
          });
        });
      }),
    );

    return new Promise<ErrorField>((resolve, reject) => {
      allValidations
        .then(() => {
          resolve({
            cid: cid.value,
            errors: [],
          });
        })
        .catch(error => {
          if (typeof error === 'string') {
            resolve({
              cid: cid.value,
              errors: [error],
            });
          } else {
            reject(error);
          }
        })
        .finally(() => {
          processing.value = false;
        });
    });
  };

  const startWatcher = () => {
    watchHandler = watch(fieldContext.modelValue, () => {
      validate().then(result => {
        form.addError(result);
      });
    });
  };

  startWatcher();

  const reset = () => {
    dirty.value = false;
    touched.value = false;
    processing.value = false;
    watchHandler(); // stop the validation watcher
    fieldContext.onReset().then(() => startWatcher());
  };

  form.registerField({cid, label, dirty, touched, processing, validate, reset});

  onBeforeUnmount(() => {
    form.unregisterField({
      cid,
      label,
      dirty,
      touched,
      processing,
      validate,
      reset,
    });
  });

  const hasError = computed(() => {
    return form.searchErrors(cid.value).length > 0;
  });

  const message = computed(() => {
    return hasError.value ? form.searchErrors(cid.value)[0].errors[0] : null;
  });

  return {
    form,
    reset,
    validate,
    hasError,
    message,
  };
}
