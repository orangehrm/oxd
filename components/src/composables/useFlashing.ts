import {ShallowReactive, ref, watch} from 'vue';

export default function useFlashing(
  props: ShallowReactive<{
    order: object;
    items: object[];
    loading: boolean;
  }>,
) {
  let oldValue: object[] = [];
  const flashIndexes = ref(Array(props.items.length).fill(false));

  // Reset cached values if loading or sorting
  watch([() => props.loading, () => props.order], () => (oldValue = []), {
    flush: 'pre',
  });

  watch(
    () => props.items,
    newValue => {
      flashIndexes.value = Array(newValue.length).fill(false);
      for (let i = 0; i < flashIndexes.value.length; i++) {
        if (oldValue.length === 0) break;
        const isNew =
          oldValue.findIndex(item => {
            const oldVal = JSON.stringify(Object.entries(item).sort());
            const newVal = JSON.stringify(Object.entries(newValue[i]).sort());
            return oldVal === newVal;
          }) === -1;
        if (isNew) flashIndexes.value[i] = true;
      }
      oldValue = JSON.parse(JSON.stringify(newValue));
    },
    {deep: true, flush: 'post'},
  );

  return flashIndexes;
}
