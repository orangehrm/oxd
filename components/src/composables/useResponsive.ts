import {nextTick, onBeforeUnmount, onMounted, reactive, toRefs} from 'vue';

interface Breakpoint {
  [screenType: string]: {
    min: number;
    max: number;
  };
}

export interface State {
  windowWidth: number;
  windowHeight: number;
  screenType: string;
}

export const breakpoints: Breakpoint = {
  xs: {
    min: 0,
    max: 599,
  },
  sm: {
    min: 600,
    max: 799,
  },
  md: {
    min: 800,
    max: 999,
  },
  lg: {
    min: 1000,
    max: 1199,
  },
  xl: {
    min: 1200,
    max: 4000,
  },
};

export default function useResponsive() {
  const state: State = reactive({
    windowWidth: 0,
    windowHeight: 0,
    screenType: '',
  });

  const setWindowSize = () => {
    const width = document.documentElement.clientWidth;
    const height = document.documentElement.clientHeight;
    state.windowWidth = width;
    state.windowHeight = height;
    for (const screenType in breakpoints) {
      if (
        breakpoints[screenType].min <= width &&
        breakpoints[screenType].max >= width
      ) {
        state.screenType = screenType;
        break;
      }
    }
  };

  onMounted(async () => {
    await nextTick();
    setWindowSize();
    window.addEventListener('resize', setWindowSize);
  });

  onBeforeUnmount(() => {
    window.removeEventListener('resize', setWindowSize);
  });

  return state;
}
