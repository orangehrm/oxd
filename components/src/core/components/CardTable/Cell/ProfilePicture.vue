<template>
  <oxd-profile-pic
    :size="profilePicture.size"
    :imageSrc="profilePicture.src"
    :link="profilePicture.link"
    :link-mode="profilePicture.target"
  />
</template>

<script lang="ts">
import {defineComponent, computed} from 'vue';
import ProfilePic from '@orangehrm/oxd/core/components/ProfilePic/ProfilePic.vue';
import {SIZE_SMALL} from '../../ProfilePic/types';
import {cellMixin} from './cell-mixin';
import {TargetTypes, TARGET_SELF, TARGETS} from './types';

export default defineComponent({
  name: 'oxd-table-cell-profile-pic',
  mixins: [cellMixin],
  components: {
    'oxd-profile-pic': ProfilePic,
  },
  props: {
    link: {
      type: String,
      default: null,
    },
    size: {
      type: String,
      default: null,
    },
    target: {
      type: String,
      default: null,
      validator: (value: TargetTypes) => {
        return TARGETS.indexOf(value) !== 1;
      },
    },
  },
  setup(props) {
    const profilePicture = computed(() => {
      return {
        src: props.item ?? null,
        size: props.size ?? SIZE_SMALL,
        link: props.link ?? null,
        target: props.target ?? TARGET_SELF,
      };
    });
    return {
      profilePicture,
    };
  },
});
</script>
