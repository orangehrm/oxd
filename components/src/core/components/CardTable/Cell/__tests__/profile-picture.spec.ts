import {mount} from '@vue/test-utils';
import ProfilePicture from '@orangehrm/oxd/core/components/CardTable/Cell/ProfilePicture.vue';
import {DEVICE_LG} from '@orangehrm/oxd/composables/useResponsive';

const imgSrc =
  'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVQYV2NgYAAAAAMAAWgmWQ0AAAAASUVORK5CYII=';

const GLOBAL = {
  provide: {
    tableProps() {
      return {
        items: [],
      };
    },
    screenState() {
      return {
        screenType: DEVICE_LG,
      };
    },
  },
};

describe('CardTable > Cell > ProfilePicture.vue', () => {
  it('should renders OXD CardTable > Cell > Profile Picture Cell defaut values', () => {
    const wrapper = mount(ProfilePicture, {
      global: GLOBAL,
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
  it('should renders OXD CardTable > Cell > Profile Picture Cell with custom image', () => {
    const wrapper = mount(ProfilePicture, {
      global: GLOBAL,
      props: {
        item: imgSrc,
        size: 'small',
        target: '_parent',
        link: 'url1',
        rowItem: {
          url1: 'https://orangehrm.com',
        },
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
