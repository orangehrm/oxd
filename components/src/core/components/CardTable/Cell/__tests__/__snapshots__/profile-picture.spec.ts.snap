// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardTable > Cell > ProfilePicture.vue should renders OXD CardTable > Cell > Profile Picture Cell defaut values 1`] = `<div class="header-image profile-image--small"><span class="img-tag" style="background-image: url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAZAAAAGQCAYAAACAvzbMAAAACXBIWXMAAA7EAAAOxAGVKw4bAAAcfUlEQVR4nO3d2XLrNtaG4cWZFDXa20n6/m+t+yDblqyJ84D/ID9V8R5tWhIJ8H2qXJ2DdPcqieJHYAGgpZRSAgDAB9lDFwAA0BMBAgDohQABAPRCgAAAeiFAAAC9ECAAgF4IEABALwQIAKAXAgQA0AsBAgDohQABAPRCgAAAeiFAAAC9ECAAgF4IEABALwQIAKAXAgQA0AsBAgDohQABAPRCgAAAeiFAAAC9ECAAgF4IEABALwQIAKAXAgQA0AsBAgDohQABAPRCgAAAenGHLgC4pbZtL39Kqct//uzvVyzLEsuy3vyzbdvf/bNt25d/BkxGgEB7XTDUdf3mr2kaaZrmTXj8KkDe49sQ+TZAvv1zXfe7P4IFprDUe385wEi0bStVVUlRFJLnuZRleQmKbwNiaN8GjOM44nmeBEFw+XMcZ+gygV4IEIxW27aXYCjLUvI8lyzLpCiKUYTDtXieJ2EYShiGEkWROI7DNBi0QIBgVJqmkaqq3owwiqKQpmmGLu0uLMsS3/clDEMJgkB83xff98VxHMIEo0OAYHBKqUtYZFkmZVlKVVXStu3QpQ3O87xLiERRJLPZTGybxZMYBwIEg2nbVo7HoyRJ8qaPgR/r+iez2UwWi4X4vs+oBIMiQHA3Simp61rKspTT6STn85nA+IQgCGS9XksURZfVXQQK7okAwc11q6ayLJPT6SR5nhvVBB+a7/sSx7HMZjMJw5BVXbgbAgQ3lee5nM9nSdPUuNVTY+O6rsxms0uYECS4NQIEN1FVlez3ezmfz1LXNcFxR47jSBiGslqtZD6fD10ODEaA4GqUUlKWpRyPRzkcDvQ3RmA2m8lms5EwDNlXgqsjQPBpXXM8SRLZ7/dSluXQJeFfHMeR+Xwui8VCoigiRHA1BAg+pW1bSZJEjsejpGnKVNWIeZ4ny+VSlsuleJ43dDkwAAGC3pqmkefnZ0mSZDI7xXVnWZZEUSSbzUZmsxmjEXwKAYIP6U61TZJEttutVFU1dEnoabVaycPDAycEozcCBB9SFIUcDgc5Ho80yQ3QNdmjKOKIFHwYAYJ3S9NUttutZFk2dCm4Is/zZLVayWq1Yu8IPoQAwW8ppeR0Oslut2OFlaFs25Y4juXPP/9kJIJ3I0DwS3Vdy+vrq+z3e1ZYTUAYhvL09CRhGNIXwW8RIPipqqpku93K8XgcuhTcURAE8vDwIHEcMxrBLxEg+KGiKGS73cr5fB66FAzA8zx5eHiQ1Wo1dCkYMQIE3ymKQp6fnyVN06FLwYAcx5GnpydZLpdDl4KRIkDwRpZl8vXrVymKYuhSMAK2bcvDw4Os12ums/Adrghc5HlOeOCNtm3l9fWVwzHxQwQIROSfaauXlxfCA99pmkZ2ux2LKfAdAgRSlqVst1t6Hvippmnk5eVFkiQZuhSMCAEycd1SXVZb4XfatpW///5bkiRhTxBEhACZNKWUvL6+yul0GroUaKKua9lut7yeGCJCgEyWUury5kDgI/I8l91uJ3VdD10KBkaATFSaprLb7XiKRC/d2ycxbQTIxCilLrvMeZcH+vr39CcPIdNFgExM27ay3+8lz/OhS4EBttst19KEESATczqdaJrjasqylP1+z2h2ogiQCanrWna7HTuKcVVJksj5fGYqa4IIkIlo21aen59ZOYOr6447YSpregiQCejeKMhOc9xK9+IxRiHTQoBMQFVVcjqdpGmaoUuBwc7nM0edTAwBYri2beV8PkuWZUOXgglgg+G0ECCG61bJMLWAeyiKQg6HA9fbRBAghjsejzwR4m6UUpIkCa8FmAgCxGBlWfIOB9xdURSSJAnLxSeAADHYfr/nR4y761b9MfI1HwFiqCzLeMcHBlOWJSceTAABYqBu5RXLdjGkw+HANWg4AsRAVVVJlmWshMGg6rpmX4jhCBDDKKUkz3NWwWAUjscjDzIGI0AM0zQNB9thNLoVWTATAWKYbvoKGIOmaSRNU1YDGooAMQzr7zE2eZ7zvhBDESCGYekkxqYsSynLcugycAMEiEHSNOVJD6PTtq2kacqSXgMRIAahWYmxyvOcADEQAWKIpmlonmO0yrJkdGwgAsQQZVly9hBGSynFGzENRIAYQCklWZYxRYBRS9OU/UmGIUAMoJSSsiz5cWLUiqLgIccwBIgB6rpmfhlayPN86BJwRQSIAeq6pv8BLRAgZiFANKeUkqqqCBBoIc9zploNQoBoTikldV3zo4QW6rqmD2IQAkRz3QgE0EHbtlyvBiFANMcPEjrhgccsBIjmuiksQAdt23K9GoQA0ZxSijllaIPr1SwEiOaapuH9H9BK0zQs+jAEAaI5nuagm7ZteegxBAGiOeaToRsCxBwEiOb4IUI3SimmsAxBgGiOHyJ0Q4CYgwDRHD9EAEMhQDRHgAAYCgGiOcuyhi4BwEQRIJqzbb5C6MWyLB58DMHdR3MECHRDgJiDu4/mXNcdugTgQ2zb5sHHEHyLmiNAoBsCxBx8i5rjxwjd2LbNFJYhuPNozrIsAgTasCxLHMcZugxcCXcezVmWxTQWtEGAmIUA0Rw/SOiE69UsBIjm+EFCJ4yYzUKAaI4fJHTCA49ZCBDNESDQiW3bXK8GIUA0163CYlkkdOA4DqsGDcI3aQDbtpkWgBYYfZiFADEAT3XQhed5Q5eAK+KuYwDHcRiBQAu+7w9dAq6IADEAx5lAF4xAzMJdxwD0QKALAsQsBIgBWFsPHTBSNg/fpgEsyxLP8/hxYtR832e5uWG44xjCdV1+nBg1pq/MQ4AYwnVdRiAYNVZgmYc7jiEYgWDsGIGYhwAxhOu6NNIxaoxAzEOAGIQfKMaKKVYz8Y0ahCkCjBVTrGYiQAzCCARj5XkeAWIgAsQgQRAMXQLwQ+xTMhPfqEGYJsBYcW2aiQAxSLcjHRiT7qw2AsQ8BIhh6INgbHjdgLkIEMOEYTh0CcAbruvyJkJDESCGYQoLY8MIxFwEiGFYLomxYROhufhWDWPbNtMFGA3LsliBZTACxDC2bdNIx2hwPZqNADEMS3kxJoyIzUaAGIYfLMaEBxqzESAGYs4ZY0ED3Wx8s4bpmpaMQjAGQRDwMGMwAsRArLvHWHDAp9kIEMO0bSt5nktd10OXAkiWZVJV1dBl4EYspZQaughcz3a7lf1+L03TDF0KIJZlyWw2ky9fvjAaMRABYoimaWS328nr6+vQpQDfCYJA/vzzT85qMwxTWAZo21aOx6McDoehSwF+qCgKeXl5YWRsGAJEc0opyfNcDoeDtG07dDnAT6VpKvv9Xpj0MAcBorlu9FGW5dClAL91PB4lz/Ohy8CVECCaS9NUjsfj0GUA71JVFYs8DEKAaI6+B3ST57kURTF0GbgCAkRjVVVJlmVDlwF8SHfd0rPTHwGisfP5TEMSWsqyjGksAxAgmlJKSZIkQ5cB9EKAmIEA0VRd1xwRAW0ppeiDGIAA0VRZlkxfQWss59UfAaKpqqpoQkJr7F3SHwGiqbquGYFAa0zB6o8A0ZBSStq2JUCgNR6C9EeAaIrpK5iAANEbAaIhpRQ/PBiB61hvBAiAwRAgeiNANGRZlliWNXQZwKfZNrcgnfHtaciyLH540B7Xsf749jRl2zajEGjNcZyhS8AnESCacl2XAIHWPM8bugR8EgGiKcdxCBBozXXdoUvAJxEgmmIEAt0xAtEfAaIpRiDQHQGiPwJEU57nsYIFWvN9f+gS8EncgTRlWRZPcNCW67o8ABmAb1BjQRAMXQLQCyNoM/ANaowAga4IEDPwDWosDMOhSwB68X2fADEA36DGXNelEQntOI4jnuexitAABIjmoigaugTgQ1zXZQGIIQgQzc1ms6FLAD7E8zwCxBAEiOaCIODHCG3Yti1RFHGQoiEIEM05jsMoBNpwHIdpV4MQIJrrnuhY0QIdhGHI8nODcNfRnGVZ/CihjeVyyeorgxAgBvB9XxaLBaMQjFoURRLH8dBl4Iq44xhiPp+zsRCj5TiObDabocvAlREghnBdVx4eHpgewCgtFgua5wYiQAwSRRFPeRidKIpktVoxxWog3ilpEMuyZLPZiFJK9vu9KKWGLgkTF4ahPD09scjDUASIYbq5Zsuy5HA4SNM0Q5eEiZrNZvLlyxd6cwazFI+pRmqaRpIkkd1uJ2VZDl0OJsRxHFkul7JarTjs03AEiMGUUlKWpZxOJ0YjuIv5fC7r9VqCIBDbtlnUYTgCZCLqupbj8SiHw0HquqY/gqvpjifZbDastJoYAmRi8jyX4/Eo5/NZ6roeuhxoLo5jWa1WMpvNWGU1QQTIBNEfwWdZliXr9VrW6zWnQU8YATJRSinJ81z++9//Mp2FD9tsNvLw8MCx7BPHmHOiLMuSKIrkjz/+oNGJD1ksFrLZbAgPECBTF8cx7xPBu7muK8vlUlyXLWQgQCbPcRyZz+eMQvAuQRCw0goXBMjEWZYlvu+z4QvvEscxq61wwZUAAgTvNp/Phy4BI0KAQBzHEc/zmMbCL7muS+8DbxAgEBEhQPBbHIqIbxEgEJF/prGY28avcCQ7vsUdAyJCgOD3GIHgW9wxICL/9EFopONnHMeh/4HvECC44AkTP9Mdzw78G1cELtgghp9hihM/whWBizAMuUngO5ZlMQLBD3FF4MKyLKax8B3XdVnmjR8iQPAGByviW57n8c4P/BABgjfCMORJE2/4vs8KLPwQAYI3PM9jGgsXjuPwUIGfIkDwRnfDAES4HvBrBAje6BrprLiByD/7P+h/4Ge4S+AN3g+Cf4vjmOkr/BQBgu/4vs+0BcS2bd7/gV8iQPAdy7JkNpuJ4zhDl4IBzedzpjLxS1wd+KEoipj7njDLsmSxWAxdBkaOAMEPOY7D9MWEhWFIHwy/RYDgp1arFVMYE9SNPtg8iN/h7oCfchxHlsvl0GXgzoIgkCiKWH2F3yJA8EuLxYJm+oR0Cyjof+E9CBD8ku/79EImxHVdVl/h3bhK8EuO48hisaChOhGr1Yo9QHg3AgS/FYYhO5InwHVdel74EAIEv2XbtiyXS+bFDWZZlmw2G1Ze4UMIELyL7/uyXq+HLgM3MpvN6HXhwwgQvItlWbJeryWO46FLwZV5niebzYYRJj6MAMGHPD4+0lA3iG3bNM7RGwGCD+mmstgbYoY4jmW5XLJsF71w1eBDbNuWxWLBfLkBHMeRx8dHGufojQDBh3U3Hvoh+nJdV/766y+mI/EpBAh6cV1Xnp6euAFpyHEc2Ww2EkXR0KVAcwQIevN9X758+cLqHY1YliXL5ZK+B66CKwifMpvNZLPZ0FTXRBzHfF+4Grpn+JRuGaiIyHa7laZpBq4IPzObzeSvv/5i5IGr4UrCp3XTIl2QYHyiKJKnpyfCA1fF1YSrsG1bNpsN79EeId/35eHhgQUPuDoCBFfjOI48PDywumdEuhVXs9mM05RxdQQIrioIAvny5QtPuyOx2WxktVoRHrgJAgRXF4ahrFYr5tsHFkURJyjjpviF4+osy5LFYsFU1oC66URCHLfE1YWbcF1XNpvN0GVM1nw+54Rd3BwBgpuZzWaclzUAz/NksViwWRA3R4DgpnjH9v2FYShBEAxdBiaAAMFNBUHAiqw7sm1bwjBk9IG7IEBwU47jMBd/R12AAPdAgOCmbNtmOuWOHMfh88bdECC4KcuyxHVdlpPeied5fNa4G6403JzjOMzJ3wn9JtwTAYKbs22bp+I74f3muCd+1bg5y7I4i+lOGOnhnggQ3BwBcj+M9HBPXG24OQLkfviccU8ECG5OKTV0CZPBZ417IkBwF9zY7oPPGfdEgAAAeiFAcHNKKZ6M74TPGfdEgODmCJD7adt26BIwIQQIbo4AuZ+maYYuARNCgODmCJD7YQSCeyJAcHNt23JjuxNGILgnAgQ3xwjkfggQ3BMBgptSSknTNIxA7qSua8Iad0OA4Ka6AMF9tG3L5427IUBwUwTIffF5454IENyUUkrquh66jMng88Y9ESC4qbZtpSzLocuYDD5v3BMBgptq21aqqhq6jMnoPm8a6bgHAgQ3VZYlN7M7q+uaVW+4CwIEN1UUxdAlTE5d1/RBcBcECG4qTdOhS5icqqqYNsRdECC4maZpGIEMoGka+iC4CwIEN5Nl2dAlTFae5/RBcHMECG4mSZKhS5isPM/ZUIibI0BwE3VdMwIZUFVVTB/i5ggQ3ESapjwBD+x0Og1dAgxHgODq2raVLMsIkIGlacpyXtwUAYKrK8tS8jwfuozJa9tWzufz0GXAYAQIrkopJVmWMf8+EofDgZEgboYAwVXVdc1T74jUdS1JkrAnBDdBgOCqsixj+mpEmqaRJEkYheAmCBBcjVJKXl9fedodmTRNJc9zvhdcHQGCq1BKyeFwoPcxQk3T0AvBTRAguIqqquRwOAxdBn4iTVMOtsTVESD4tLZt5Xg88ia8EVNKyW63YxSCqyJA8Gl5nsvpdGKOfeTKspT9fj90GTAIAYJPqapKdrsd75/QxH6/J+xxNQQIeusa58yt66NpGtnv97xqGFdBgKC3JEmYEtFQnuey3+95Xwg+jQBBL0VRyHa75SakIaWUHI9HTuvFpxEg+BCllJRlKc/Pz+z50JhSSr5+/coxJ/gUAgQfUte17HY7+h6G+Pr1q2RZRoigFwIE79btJWDqwxxVVcnLywtHnaAXAgTvopSS7XYrx+ORG41h8jyXl5cXlmLjwyzF3QC/oJSSpmlkt9vJ4XAgPAzm+7785z//Ed/3xbKsocuBBhiB4JcIj+koy1L+/vtvprPwboxA8FNN08jz8zM7lycmDEN5fHyU2WzGSAS/RIDgh6qqkufnZ94uOFGe58mXL19ksVgMXQpGjADBG23bSpZl8vLywj4PyNPTkyyXS7Ftm9EIvkOA4KI7lv319ZUVORAREcuyZLlcynq9prmO7xAgEJF/pqz2+70cDgeOJ8EblmVJGIby8PAgcRwPXQ5GhACZOKWUFEUhLy8v7EjGL7muK4+Pj7JcLhmJQEQIkMnq9nccj0fZ7XaMOvBu8/lcHh4eJAgCgmTiCJAJattW8jyXw+HAsSToxfd9Wa1WslwuxXGcocvBQAiQianr+hIcvMMcn2FZliwWC1mtVhKGIaORCSJAJiTLsssx7HztuBbP82S1WslqtWI0MjEEiOHatpWiKC6jDr5u3IrnefL09CRRFLFvZCIIEEMppaSqKjmdTnI8HtnXgbvoprUWi8UlSGAuAsRAdV3L+XyW8/nMi58wCM/zZDabyXq9liAIhi4HN0KAGOZ0Osl+v5eiKFiai8H5vn9ptLuuO3Q5uDICRHPdfo4sy2S/30uWZUOXBHzHcZzLsl/XdZnaMgQBorG6riVN08tUFSMOjF0QBLJYLGQ2m7ER0QAEiIaappHz+SxJkhAc0I5lWeL7vsxmM1mtVuL7/tAloScCRCNKKTkej3I8HulxQHuWZYnneRLHsaxWK/E8jxGJZgiQkWvbVqqqkiRJ5HA4sBwXRrIsS+bzuazXa/E8TxzHIUw0QICMVNM0UhSFJEkiSZJw7AgmwbZtieNY4jiWMAyZ3ho5AmRkmqa59DayLGPEgUmybVuCILiECS+zGicCZCS6zX+Hw0HqupamaYYuCRhc1yeJokiWy6VEUTR0SfgXAmQg3f6NsizleDzK+XymKQ78RhRFl9N/2U8yPALkztq2lbIsJc9zSdNUkiThgEPgg3zflziOL/tJaLoPgwC5k263+Pl8lqIopCxLggP4JNd1JQxDCcNQ4jjm3K07I0BurCxLOZ1OkiSJVFVFbwO4AcuyxHXdS68kDEOmt+6AALmypmmkaRrJ8/wSHADuy/M8mc/nslgsxHVdprhuhAC5AqWU1HUtRVFImqaSpin7NoARsG1bZrOZzGazy74SRibXQ4B8glJKsiyTNE0lz3MpioIpKmCEuvO3wjCU2WwmURQxKrkCAqSHbrPf6XS6hAYfI6AHx3HEdV0Oc7wCAuQ3lFLf7dlIkoQ9G4Ahoii6vIK321vCyOR9CJBf6Jrh3Z6NPM8ZaQCGcl1X5vP5ZVkwpwP/HgHyDaWUVFV1OYsqyzKp63rosgDcSXcOVxiGMp/PJQgCGu8/QYD8S1EUl9fC1nXNNBUwYd3eEt/3ZblcShzHBMk3Jh0g3bs2siyT4/EoeZ4PXRKAkere6x7HMe8s+X+TC5CuIZ7n+eVoEY5MB/Be3d6SOI4liqJJ90omFSBVVcn5fJY0TaUoCnobAHqzLOvNOVxTPD7F+ABRSr1ZfktvA8A1WZYljuNIEASyXq9lNptNZkRiXID8e99GnueXpjgA3IPneZemu+d5Ru8rMSpAlFJvehs0xQEMpdtX0p3DZWLT3YgAadtWsiy7vEucgwwBjIXrupf3u8/nc3Fdd+iSrkbrAFFKyfl8luPxSFMcwKjZti2+78t8PpflcmlEkGgXIN3ejTRN5XA4MNoAoB3btmU+n8t6vda6T6JNgLRtK0VRSJIkcj6fCQ4A2nMcR+I4vuwp0W1UMvoAob8BwHSO40gYhrJYLGQ2m2kTJKMOkCzL5PX1VfI8p78BwHiO44jv+7JYLGS5XI5+Y+KoAqTbv5FlGfs3AEya4zjy8PAgcRyL67piWdbo+iSjCJDuneJ5nsvxeJQ0TXnvBgCISBAEMp/PJY5jCYJgVCEyeIDUdS3n81mSJJEsyzhmBAB+oAuS+Xwuvu+PIkgGCxCllCRJIrvdTsqyJDgA4DcsyxLP8y5LgIdutt81QLoeR5qmst/vOWoEAHqybVseHh4uu9uHaLjfLUDqur68uIkeBwBcRxAEslgsLj2Se7p5gLRtK2mayul0kjRNpWmaW/7fAcDkWJYlQRDIcrm863lbNw2Qsizl5eVFsiwjOADgxmzbvryX5B7vcL96gPz7BU6Hw4HmOAAMIIoi+fLli/i+f7Oztq4aIN0rY7vTcQEAw3EcRxaLhSwWCwmC4OojkqsESNu2kiQJDXIAGKHuGPlrL/39dICUZSm73U6SJKHPAQAjZVmWhGEom81G4ji+ypRWrwDp9nOcTid5eXlhxAEAGpnP57LZbCQMw08FyYcDpFuWezgcmK4CAE25rns5rNHzvF7/Gx8KkLquZbfbyfl85nh1ANCcbdsSx7GsViuJoujDo5F3BYhSSrIsk91uJ1mWMeoAAIO4rivr9VoeHh4+9N/7bYBUVSWHw0H2+z17OgDAYHEcy+Pj42XvyO/8NECUUlIUhWy3W0mS5OqFAgDGx/d92Ww2Mp/PxXGcX/67PwyQtm3lfD5zYi4ATFC3AfHx8fGXIfJdgDRNI/v9Xvb7Pfs6AGCiLMuSOI4vx6H88N/pAqR7rezXr1+ZsgIAiMg/U1qPj48/PJzRUv+Qoijk5eVF0jQdqEwAwBh1e0ZWq9Wbpb5W27YqyzLZbreSZdmAJQIAxsq2bXl8fJTVanUZibhFUcjz8zOn5wIAfqptW9ntdqKUkvV6LbZti/u///2PZjkA4LeappHtdnsJEZfwAAC8l1JKXl9fxbZtue37DgEAxummswgQAMCHNU1DgAAA+iFAAAC9ECAAgF4IEABALwQIAKAXAgQA0AsBAgDohQABAPRCgAAAeiFAAAC9ECAAgF4IEABALwQIAKAXAgQA0AsBAgDohQABAPRCgAAAeiFAAAC9ECAAgF4IEABALwQIAKAXAgQA0AsBAgDohQABAPRCgAAAeiFAAAC9ECAAgF4IEABALwQIAKAXAgQA0AsBAgDohQABAPTyf5VPed5l1/xgAAAAAElFTkSuQmCC);"></span></div>`;

exports[`CardTable > Cell > ProfilePicture.vue should renders OXD CardTable > Cell > Profile Picture Cell with custom image 1`] = `<div class="header-image profile-image--small"><a href="https://orangehrm.com" target="_parent"><span class="img-tag" style="background-image: url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVQYV2NgYAAAAAMAAWgmWQ0AAAAASUVORK5CYII=);"></span></a></div>`;
